Dockerfile

```diff
@@ -1,25 +1,31 @@
 FROM python:3.6-slim
 MAINTAINER Anthony Miyaguchi <amiyaguchi@mozilla.com>
 
-# bootstrap
 ENV PYTHONUNBUFFERED=1 \
+    PIPENV_VENV_IN_PROJECT=1 \
+    # AWS_ACCESS_KEY_ID= \
+    # AWS_SECRET_ACCESS_KEY= \
+    SHELL=/bin/bash \
     PORT=8000
 
 EXPOSE $PORT
 
+# Bootstrap the system with root privileges
 RUN apt-get update && \
     apt-get --yes install make git
 RUN pip install --upgrade pip
 RUN pip install pipenv
 
+# Create the application user
 WORKDIR /app
 RUN groupadd --gid 10001 app && \
     useradd --gid app --uid 10001 --home-dir /app app
 RUN chown -R app:app /app
 
+# Start the userland environment
 COPY . /app
-
-# start userland
 USER app
+ENV PATH="/app/.venv/bin:$PATH"
+
 RUN make sync
```

Comment:
Perhaps a peculiarity of my environment (Docker 1.13.1-51.git4032bd5 on Fedora, umask 0002), the copy resulting from this command while running in the container fails because of bad permissions (I ran `make build`):

    Copying testing schemas                                                                                                                                                                                                                                                                                                                                    
    uid=10001(app) gid=10001(app) groups=10001(app)
    # file: tests/resources/schemas
    # owner: root
    # group: root
    user::rwx
    group::r-x
    other::r-x
    
    # file: resources/schemas
    # owner: root
    # group: root
    user::rwx
    group::rwx
    other::r-x
    
    ‘tests/resources/schemas/testing/test/test.1.schema.json’ -> ‘resources/schemas/testing/test/test.1.schema.json’
    cp: cannot create regular file ‘resources/schemas/testing/test/test.1.schema.json’: Permission denied
    Makefile:2: recipe for target 'sync' failed

This isn't related to this change set, and the code clearly works in e.g. the circle build environment. I can add `USER root` before the make sync to fix it, which I will use as a workaround for now.

Reply:
I am reproducing the same issue on Fedora with Docker version 1.13.1, build `7f1fa5c-unsupported`. 

The error is slightly different now that I use `rsync` instead of `cp`. 

```
INCLUDE_DATA=false bash sync.sh
Copying schemas from ./mozilla-pipeline-schemas/schemas
~ ~
commit 1dfa9f01194784a59b0513395463ede9c33e4c54
Author: Mark Reid <mreid@mozilla.com>
Date:   Wed May 30 13:13:52 2018 -0300

    Add a .dockerignore file to skip the release dir. (#164)
~
sending incremental file list
cannot delete non-empty directory: testing/test
cannot delete non-empty directory: testing
./
rsync: failed to set times on "/app/resources/schemas/.": Operation not permitted (1)
rsync: delete_file: unlink(testing/test/test.1.schema.json) failed: Permission denied (13)
rsync: delete_file: rmdir(testing/test) failed: Permission denied (13)

sent 2.67K bytes  received 410 bytes  6.17K bytes/sec
total size is 395.76K  speedup is 128.37
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1183) [sender=3.1.1]
make: *** [sync] Error 23
Makefile:2: recipe for target 'sync' failed
The command '/bin/sh -c make sync' returned a non-zero code: 2
make: *** [Makefile:13: build] Error 2
```